src/parsers/strategies.rs

```diff
@@ -261,6 +251,24 @@ fn transformer(column: ColumnInFile, overrides: &TransformerOverrides) -> Transf
     apply_transformer_overrides(column.data_category, overrides, column.transformer)
 }
 
+/// Validates transformers with deterministic=true have an id_column (except FakeUUID transformer).
+fn validate_deterministic_settings(strategy: &StrategyInFile, errors: &mut ValidationErrors) {
+    for column in &strategy.columns {
```

Comment:
There are currently many if statements in this function. It would be better to use filter instead of writing multiple if conditions. This will improve readability

Reply:
I've addressed this